index.js

```diff
@@ -13,133 +13,167 @@
+    return args.map(argument => {
+      if (argument.type !== "ObjectExpression") {
+        return argument;
+      }
+      return {
+        ...argument,
+        properties: argument.properties.map(prop => {
+          if (prop.key.name !== "content") {
+            return prop;
+          }
 
-    return {
-        visitor: {
-            /**
-             * Find all JSX elements that use the old
-             * <glamorous.Something> notation and rewrite them to be
-             * plain ole somethings.
-             */
-            JSXElement(path) {
-                path.traverse({
-                    "JSXOpeningElement|JSXClosingElement"(path) {
-                        if (!t.isJSXMemberExpression(path.node.name)) {
-                            return;
-                        }
-                        if (path.node.name.object.name !== "glamorous") {
-                            return;
-                        }
-                        path.node.name = t.identifier(
-                            path.node.name.property.name.toLowerCase()
-                        );
-                    }
-                });
-            },
+          // Add quotes to the content property.
+          return {
+            ...prop,
+            value: t.stringLiteral(`"${prop.value.value}"`),
+          };
+        }),
+      };
+    });
+  };
 
-            /**
-             * Find glamorous import statements, including ThemeProvider
-             * imports and rewrite them to be emotion imports.
-             */
-            ImportDeclaration(path, { opts }) {
-                if (path.node.source.value !== "glamorous") {
-                    return;
-                }
+  const transformJSXAttributes = attrs => {
+    if (!attrs) return [];
+    const appendToCss = [];
+    let cssAttr = null;
+    const newAttrs = attrs.filter(attr => {
+      const {value, name, type} = attr;
+      if (type === "JSXSpreadAttribute") return true;
+      if (name.name === "css") {
+        cssAttr = attr;
+      } else {
+        // event handlers are no CSS Props!
+        if (name.name.indexOf("on") === 0) return true;
 
-                // First, check for ThemeProvider and update that.
-                const themeProviderIndex = path.node.specifiers.findIndex(
-                    specifier => specifier.local.name === "ThemeProvider"
-                );
+        appendToCss.push({
+          name: t.identifier(name.name),
+          value: value.type === "JSXExpressionContainer" ? value.expression : value,
+        });
+        return false;
+      }
+      return true;
+    });
+    if (appendToCss.length > 0) {
+      if (!cssAttr) {
+        cssAttr = t.jsxAttribute(
+          t.jsxIdentifier("css"),
+          t.jsxExpressionContainer(t.objectExpression([]))
+        );
+        newAttrs.push(cssAttr);
+      } else if (cssAttr.value.expression.type !== "ObjectExpression") {
+        // turn <span css={obj} .../> into <span css={{...obj}} .../>
```

Comment:
I think the intent here is to turn `<span css={obj()} into <span css={{ ...obj() }}`? The current description of spreading an object into an object seems to be a bit of a no-op. 😅

Reply:
This code is needed to transform e.g.

```<gl.Span css={redStyle} marginTop={5}/>```

into

```<span css={{...redStyle, marginTop: 5}}/>```

So the comment doesn't tell the whole story yet, I'll fix it!